### PR TITLE
Wh/remove channels fix

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Removes channels that just Wait and phase Reset since these are doing nothing!

### CODE REVIEW

File: tests/qat/test_quantum_backend.py [hunk 1]
```diff
@@ -632,3 +638,28 @@ def test_execution_batching(self, repeat_count, repeat_limit, expected_batches):
...
         generated_batches = engine._generate_repeat_batches(repeat_count)
         assert generated_batches == expected_batches
+
+    def test_empty_channels_removed(self):
+        hw = get_test_model()
+        qubit = hw.get_qubit(0)
+
+        builder = (
+            get_builder(hw).X(qubit, np.pi / 2.0).measure_mean_z(qubit).synchronize(qubit)
+        )
```
Comment (hamidelmaazouz) [inline tests/qat/test_quantum_backend.py:648]: not enough testing, add more builders that involve for example a variety of pulse channels: drive, measure, acquire, second state, ...

File: src/qat/purr/compiler/execution.py [hunk 2]
```diff
@@ -468,7 +468,10 @@ def create_duration_timeline(self, instructions: List[QuantumInstruction]):
         # Strip timelines that only hold delays, since that just means nothing is
         # happening on this channel.
         for key, timeline in dict(results).items():
```
Comment (hamidelmaazouz) [inline src/qat/purr/compiler/execution.py:470]: minor: instead of a new dict:

```
dict = {k, v for k, v in dict.items() if somthing_about(k, v)}
```

**Review by hamidelmaazouz — COMMENTED** [review]
Happy with the solution, just needs more testing

File: tests/qat/test_quantum_backend.py (hunk 1 above)
Comment (keriksson-rosenqvist) [reply tests/qat/test_quantum_backend.py:648]: @willh727-OQC

Comment (willh727-OQC) [reply tests/qat/test_quantum_backend.py:648]: They are all added when you add a qubit! Do I need more tests for just check these channels?

File: src/qat/purr/compiler/execution.py (hunk 2 above)
Comment (willh727-OQC) [reply src/qat/purr/compiler/execution.py]: Done, with the formatting it does take up more lines tho

**Review by hamidelmaazouz — APPROVED** [review]

**Pull request merged**